clients/src/main/java/org/apache/kafka/clients/producer/internals/DefaultPartitioner.java

```diff
@@ -33,16 +29,17 @@
  * <ul>
  * <li>If a partition is specified in the record, use it
  * <li>If no partition is specified but a key is present choose a partition based on a hash of the key
- * <li>If no partition or key is present choose a partition in a round-robin fashion
+ * <li>If no partition or key is present choose the sticky partition that changes when the batch is full.
  */
 public class DefaultPartitioner implements Partitioner {
 
-    private final ConcurrentMap<String, AtomicInteger> topicCounterMap = new ConcurrentHashMap<>();
+    private final StickyPartitionCache stickyPartitionCache = new StickyPartitionCache();
 
     public void configure(Map<String, ?> configs) {}
 
     /**
      * Compute the partition for the given record.
+     * This method has been deprecated for DefaultPartitioner in favor of computePartition.
```

Comment:
We're not doing `computePartition` any more, right?  This comment is out of date.